### ISSUE TITLE
handle "parse" errors

### ISSUE THREAD

**Issue by sebseb7** [socketio/socket.io-client#1554]
when the code inside a .on('someevent',handler) handler fails, a "parse" error occurs, which is silently dropped and the connection is closed.

Solution: have a on('parse_error',data,error) event.

code from manager.ts which silently drops the "e" :
```
  /**
   * Called with data.
   *
   * @private
   */
  private ondata(data): void {
    try {
      this.decoder.add(data);
    } catch (e) {
      this.onclose("parse error");
    }
  }
```

**Comment by m4heshd** [comment]
Wasted so much time on this. Couldn't find WTH was going on because the error was silently consumed by socket.io client.

**Comment by darrachequesne** [comment]
This should be fixed by https://github.com/socketio/socket.io-client/commit/2403b88057bf3fd32eb2047c82be26c455c13a2f, included in version [4.5.3](https://github.com/socketio/socket.io-client/releases/tag/4.5.3). Sorry for the mess!
